Pull in r217454 from upstream llvm trunk (by Ed Maste):

  Use armv6k default for FreeBSD/ARM

  Patch by Andrew Turner.

Introduced here: http://svnweb.freebsd.org/changeset/base/269387

Index: lib/Support/Triple.cpp
===================================================================
--- lib/Support/Triple.cpp
+++ lib/Support/Triple.cpp
@@ -963,6 +963,7 @@ const char *Triple::getARMCPUForArch(StringRef MAr
     MArch = getArchName();
 
   switch (getOS()) {
+  case llvm::Triple::FreeBSD:
   case llvm::Triple::NetBSD:
     if (MArch == "armv6")
       return "arm1176jzf-s";
